File: Easy/LC1528/LC1528.java
class Solution {
    public String restoreString(String s, int[] indices) {
        int len = s.length();
        if (len == 1) return s;
        char[] arr = new char[len];
        for(int i=0; i<len; i++) {
            char c = s.charAt(i);
            arr[indices[i]] = c;
        }
        StringBuilder sb = new StringBuilder();
        for(char c : arr) {
            sb.append(c);
        }
        return sb.toString();
    }
}

File: Easy/LC1528/LC1528.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
const restoreString = (s, indices) => (
    indices.reduce((acc, curr, idx) => {
        acc[curr] = s[idx];
        return acc;
    }, []).join("")
);

// Brute Force
// const restoreString = (s, indices) => {
//     const len = s.length;
//     if(len <= 1) return s;
//     let idx = 0;
//     let arr = [];
//     for(let c of s) {
//         arr[indices[idx++]] = c;
//     }
//     return arr.join('');
// };